django/contrib/admin/widgets.py

```diff
@@ -149,7 +149,7 @@ def get_context(self, name, value, attrs):
             params = self.url_parameters()
             if params:
                 related_url += '?' + '&amp;'.join(
-                    '%s=%s' % (k, v) for k, v in params.items(),
+                    '%s=%s' % [(k, v) for k, v in params.items()],
```

Comment:
This is clearly incorrect:
* This will blow up if you don't have exactly two parameters.
* The issue is due to the generator in the arguments to `str.join()`.
* Why a list comprehension? I think it should remain a generator expression.

A correct fix would be:
```python
('%s=%s' % (k, v) for k, v in params.items()),
```

Reply:
Yeah you're right. Sorry for the wrong fix here, I think this is exactly the reason why Python 3.7 disallows this construct.